### PR TITLE
Feature/zabbix

### PR DESCRIPTION
**Pull request by tbaka**
Adds Ansible rewrite of Zabbix script.

### CODE REVIEW

File: apps/linode-marketplace-zabbix/roles/zabbix/tasks/main.yml
Comment (hmorris3293) [inline apps/linode-marketplace-zabbix/roles/zabbix/tasks/main.yml:1]: We need to figure out how to get around the setup after login. Once deployed it brings you to the url, user has to  finish the install where it asks database info. Appears that the conf gets updated /etc/zabbix/web/zabbix.conf.php so we may be able to prefil that file to get around setup page

File: apps/linode-marketplace-zabbix/provision.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      set_fact:
+        zabbix_db_pass: "{{ lookup('password', '/dev/null length=45 chars=ascii_letters,digits') }}"
+      no_log: true
+
+    - name: writing sudo user password to vars
+      lineinfile:
+        insertafter: EOF
+        path: group_vars/linode/vars
+        line: | 
+          zabbix_db_pass: {{ zabbix_db_pass }}
```
Comment (n0vabyte) [inline apps/linode-marketplace-zabbix/provision.yml:26]:
```suggestion
    - name: writing zabbix database password to vars
      lineinfile:
        insertafter: EOF
        path: group_vars/linode/vars
        line: | 
          zabbix_db_pass: {{ lookup('password', '/dev/null length=45 chars=ascii_letters,digits') }}
```

Comment (n0vabyte) [reply apps/linode-marketplace-zabbix/provision.yml:26]: we don't have to set_fact to just write the creds to the vars file. We can just write directly.

**Review by n0vabyte — CHANGES_REQUESTED** [review]

File: apps/linode-marketplace-zabbix/roles/zabbix/tasks/main.yml
Comment (tbaka) [reply apps/linode-marketplace-zabbix/roles/zabbix/tasks/main.yml:1]: updated and added generated Admin pass

File: apps/linode-marketplace-zabbix/roles/zabbix/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+- name: set zabbix site name in conf
+  lineinfile:
+    path: /etc/zabbix/nginx.conf
+    regex: '^\#        server\_name     example\.com\;'
+    line: "           server_name     {{ _domain }};"
+
+- name: update zabbix Admin password
+  community.mysql.mysql_query: 
+    login: zabbix
```
Comment (n0vabyte) [inline apps/linode-marketplace-zabbix/roles/zabbix/tasks/main.yml:78]:
```suggestion
```

Comment (n0vabyte) [reply apps/linode-marketplace-zabbix/roles/zabbix/tasks/main.yml:78]: Playbook fails because of this param.

**Review by n0vabyte — CHANGES_REQUESTED** [review]

File: apps/linode-marketplace-zabbix/roles/zabbix/templates/vhost_ssl.conf.j2 [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    index index.php;
+
+    location = /favicon.ico {
+        access_log off;
+        log_not_found off;
+    }
+
+    location / {
+        include proxy_params;
+        proxy_pass http://127.0.0.1:8080;
```
Comment (n0vabyte) [inline apps/linode-marketplace-zabbix/roles/zabbix/templates/vhost_ssl.conf.j2:45]:
```suggestion
        #proxy_pass http://127.0.0.1:8080;
```

File: apps/linode-marketplace-zabbix/roles/zabbix/templates/vhost_ssl.conf.j2 [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    {% if subdomain != "www" or default_dns is defined %}
+    server_name {{ _domain }};
+    {% elif subdomain == "www" %}
+    server_name {{ _domain }} www.{{ _domain }};
+    {% endif %}
+    location = /favicon.ico { access_log off; log_not_found off; }
+    root /usr/share/zabbix;
+    location / {
+    include proxy_params;
+    proxy_pass http://127.0.0.1:8080;
```
Comment (n0vabyte) [inline apps/linode-marketplace-zabbix/roles/zabbix/templates/vhost_ssl.conf.j2:12]:
```suggestion
    #proxy_pass http://127.0.0.1:8080;
```

**Review by n0vabyte — CHANGES_REQUESTED** [review]
We don't have to proxy_pass to 8080 because this is already running the application. We should add a task to rename `/etc/nginx/conf.d/zabbix.conf{,.dis}`. Other than that, this is solid!

**Review by n0vabyte — APPROVED** [review]
Deployment passed after requested changes. :+1:

**Pull request merged**